File: Unfinished/005.R
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
prime_base <- 2*3*5*7*11*13*17*19
num_vec = rep(c(prime_base), times = 20)
divisors = c(1:20)
zero.v = rep(c(0), times = 20) 
truth.v = rep(c(TRUE), times = 20) 
# while () {
#   num_vec = num_vec + rep(c(prime_base), times = 20)
#   print(num_vec)
# }
#
repeat {
  if ((num_vec %% divisors == zero.v) == truth.v) {
    print(num_vec)
    break
  }
  num_vec = num_vec + rep(c(prime_base), times = 20)
}

File: SolutionsPython/005.py
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
def is_divisible(guess):
    i = 20
    while i > 1:
        if guess % i != 0:
            return False
        i -= 1
    return True


def smallest_common_multiple():
    primeBase = 2 * 3 * 5 * 7 * 11 * 13 * 17 * 19
    i = primeBase

    while is_divisible(i) == False:
        i += primeBase

    return i


print(smallest_common_multiple())
# 232792560
# Runtime: 0.040s